### PR TITLE
Improve syntax error output

### PR DESCRIPTION
**Pull request by m1ranto**
Improve syntax error output on array decomposition :
- on `(a, b), (d) = fruits_vegetables` decomposition, add a new syntax error output
- on `((a, b), (d)) = fruits_vegetables` decomposition, move the syntax error output below

For two decomposition, having two distinct syntax errors outputs are more clear rather than having only one syntax error output.

### CODE REVIEW

**Review by kotp — APPROVED** [review]

File: exercises/concept/locomotive-engineer/.docs/introduction.md [hunk 1]
```diff
@@ -115,10 +115,12 @@ If the decomposition has variables with incorrect placement and/or an incorrect
 
 ```ruby
 fruits_vegetables = [["apple", "banana"], ["carrot", "potato"]]
+
 (a, b), (d) = fruits_vegetables
+syntax error, unexpected '=', expecting '.' or &. or :: or '['
```
Comment (iHiD) [inline exercises/concept/locomotive-engineer/.docs/introduction.md:120]:
```suggestion
# syntax error, unexpected '=', expecting '.' or &. or :: or '['
```

File: exercises/concept/locomotive-engineer/.docs/introduction.md [hunk 2]
```diff
@@ -115,10 +115,12 @@ If the decomposition has variables with incorrect placement and/or an incorrect
...
 
-syntax error, unexpected ')', expecting '.' or &. or :: or '['
 ((a, b), (d)) = fruits_vegetables
+syntax error, unexpected ')', expecting '.' or &. or :: or '['
```
Comment (iHiD) [inline exercises/concept/locomotive-engineer/.docs/introduction.md:123]:
```suggestion
# syntax error, unexpected ')', expecting '.' or &. or :: or '['
```

**Review by iHiD — COMMENTED** [review]
Let's add comments in while we're here.

**Review by iHiD — APPROVED** [review]

**Comment by Benyam12** [comment]
در تاریخ دوشنبه ۴ سپتامبر ۲۰۲۳،‏ ۱۱:۵۴ Miranto ***@***.***>
نوشت:

**Comment by kotp** [comment]
> در تاریخ دوشنبه ۴ سپتامبر ۲۰۲۳،‏ ۱۱:۵۴ Miranto ***@***.***> نوشت:
> […](#)
> Improve syntax error output on array decomposition : - on (a, b), (d) = fruits_vegetables decomposition, add a new syntax error output - on ((a, b), (d)) = fruits_vegetables decomposition, move the syntax error output below For two decomposition, having two distinct syntax errors outputs are more clear rather than having only one syntax error output. 

There are one each, and so two in total, and they are distinct.

Is there a different change you would like to see?

Please feel free to post and discuss in the forums.  This merge request has been accepted and merged.

You can use [this link](https://forum.exercism.org/new-topic?title=Improve%20syntax%20error%20output&body=Improve%20syntax%20error%20output%20on%20array%20decomposition%20:%0D%0A-%20on%20%60(a,%20b),%20(d)%20=%20fruits_vegetables%60%20decomposition,%20add%20a%20new%20syntax%20error%20output%0D%0A-%20on%20%60((a,%20b),%20(d))%20=%20fruits_vegetables%60%20decomposition,%20move%20the%20syntax%20error%20output%20below%0D%0A%0D%0AFor%20two%20decomposition,%20having%20two%20distinct%20syntax%20errors%20outputs%20are%20more%20clear%20rather%20than%20having%20only%20one%20syntax%20error%20output.&category=ruby) and let us know what you are thinking!

**Pull request merged**